### PR TITLE
Upgrade to ruby 3

### PR DESCRIPTION
**Pull request by bkjohnson**
The default ruby version on my system is 3.x, and I was [having problems on Ubuntu 22 getting rvm and rbenv to work](https://churchoffoxx.net/2022/05/25/rvm-cannot-compile-on-ubuntu-2204.html) due to their reliance on outdated dependencies. I tried compiling openssl myself, then switching to `rbenv`, but still was having problems so I decided to upgrade the repo instead of download older dependencies on my system.

## Testing

- local build
- deploying to dev-design.va.gov

I looked around on a few different pages and didn't notice anything broken, so I think this should be good. Feel free to checkout and try it locally.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -33,3 +33,5 @@ gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 # Performance-booster for watching directories on Windows
 gem "wdm", "~> 0.1.0" if Gem.win_platform?
 
+gem "webrick"
```
Comment (bkjohnson) [inline Gemfile:36]: Without this jekyll was failing to build for me.

**Comment by jamigibbs** [comment]
@bkjohnson Which machine did you test this on? I'm running into a gem installation error and I want to confirm if it's related to my machine or not (M1 Macbook).

**Comment by bkjohnson** [comment]
I ran it on an HP Elitebook 845 running Ubuntu 22.04. What error did you get?

**Comment by jamigibbs** [comment]
> What error did you get?

Initially, I suspected it was an issue with `openssl` but the validation for that appears to be ok looking at the log below. The first error occurs at `eventmachine` with `error: use of undeclared identifier 'rb_thread_select';`. 

- Apple M1 Pro
- macOS Monterey

```
➜  vets-design-system-documentation git:(ruby-3) ✗ bundle update
Fetching gem metadata from https://rubygems.org/............
Resolving dependencies...
Using rake 13.0.6
Using public_suffix 4.0.7
Using bundler 2.2.22
Using colorator 1.1.0
Using concurrent-ruby 1.1.10
Fetching eventmachine 1.2.7
Using http_parser.rb 0.8.0
Using ffi 1.15.5
Using forwardable-extended 2.6.0
Using google-protobuf 3.21.4
Using rb-fsevent 0.11.1
Using rexml 3.2.5
Using liquid 4.0.3
Using addressable 2.8.0
Using mercenary 0.4.0
Using sassc 2.4.0
Using rb-inotify 0.10.1
Using kramdown 2.4.0
Using pathutil 0.16.2
Using rouge 3.29.0
Using i18n 1.12.0
Using listen 3.7.1
Using kramdown-parser-gfm 1.1.0
Using safe_yaml 1.0.5
Using posix-spawn 0.3.15
Using webrick 1.7.0
Using jekyll-sass-converter 2.2.0
Using unicode-display_width 1.8.0
Fetching sass-embedded 1.54.0
Using jekyll-watch 2.2.1
Using terminal-table 2.0.0
Installing sass-embedded 1.54.0 with native extensions
Installing eventmachine 1.2.7 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/Users/jamigibbs/.rvm/gems/ruby-3.0.2/gems/eventmachine-1.2.7/ext
/Users/jamigibbs/.rvm/rubies/ruby-3.0.2/bin/ruby -I
/Users/jamigibbs/.rvm/rubies/ruby-3.0.2/lib/ruby/3.0.0 -r
./siteconf20220728-21315-aq2rl3.rb extconf.rb
--with-openssl-dir\=/usr/local/Cellar/openssl@3/3.0.5
-----
Using OpenSSL from pkg-config -I/usr/local/Cellar/openssl@3/3.0.5/include  &&
-L/usr/local/Cellar/openssl@3/3.0.5/lib && -lssl -lcrypto
-----
checking for -lcrypto... yes
checking for -lssl... yes
checking for openssl/ssl.h... yes
checking for openssl/err.h... yes
checking for rb_trap_immediate in ruby.h,rubysig.h... no
checking for rb_thread_blocking_region()... no
checking for rb_thread_call_without_gvl() in ruby/thread.h... no
checking for rb_thread_fd_select()... no
checking for rb_fdset_t in ruby/intern.h... yes
checking for rb_wait_for_single_fd()... no
checking for rb_enable_interrupt()... no
checking for rb_time_new()... no
checking for inotify_init() in sys/inotify.h... no
checking for __NR_inotify_init in sys/syscall.h... no
checking for writev() in sys/uio.h... yes
checking for pipe2() in unistd.h... no
checking for accept4() in sys/socket.h... no
checking for SOCK_CLOEXEC in sys/socket.h... no
checking for sys/event.h... yes
checking for sys/queue.h... yes
checking for clock_gettime()... yes
checking for CLOCK_MONOTONIC_RAW in time.h... yes
checking for CLOCK_MONOTONIC in time.h... yes
CXXFLAGS= -Wall -Wextra -Wno-deprecated-declarations -Wno-ignored-qualifiers
-Wno-unused-result -Wno-address
creating Makefile

current directory:
/Users/jamigibbs/.rvm/gems/ruby-3.0.2/gems/eventmachine-1.2.7/ext
make DESTDIR\= clean

current directory:
/Users/jamigibbs/.rvm/gems/ruby-3.0.2/gems/eventmachine-1.2.7/ext
make DESTDIR\=
compiling binder.cpp
compiling cmain.cpp
compiling ed.cpp
compiling em.cpp
em.cpp:706:13: error: use of undeclared identifier 'rb_thread_select'; did you
mean 'rb_thread_fd_select'?
        if ((ret = rb_thread_select(kqfd + 1, &fdreads, NULL, NULL, &tv)) < 1) {
                   ^~~~~~~~~~~~~~~~
                   rb_thread_fd_select
/Users/jamigibbs/.rvm/rubies/ruby-3.0.2/include/ruby-3.0.0/ruby/internal/intern/select.h:48:5:
note: 'rb_thread_fd_select' declared here

[...]

```

And then it attempts to install the gem again but that fails. Manually running it based on the feedback also fails:

```
Gem files will remain installed in
/Users/jamigibbs/.rvm/gems/ruby-3.0.2/gems/eventmachine-1.2.7 for inspection.
Results logged to
/Users/jamigibbs/.rvm/gems/ruby-3.0.2/extensions/arm64-darwin-21/3.0.0/eventmachine-1.2.7/gem_make.out

An error occurred while installing eventmachine (1.2.7), and Bundler
cannot continue.
Make sure that `gem install eventmachine -v '1.2.7' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  minima was resolved to 2.5.1, which depends on
    jekyll-feed was resolved to 0.16.0, which depends on
      jekyll was resolved to 4.2.2, which depends on
        em-websocket was resolved to 0.5.3, which depends on
          eventmachine


Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
```

There might be signifigance to that path `arm64-darwin-21` which is related to the M1 chip architecture.  I might need to investigate how to switch this to `x86_64`.

**Comment by bkjohnson** [comment]
This looks similar to the [error I was getting a few months ago](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/pull/679#pullrequestreview-928279580) when we were doing a previous ruby upgrade. Unfortunately I don't have a great solution for it - I ended up re-installing my OS for other reasons and didn't have the problem afterwards.

I can move this back into draft and look at other options.

**Comment by jamigibbs** [comment]
@bkjohnson The issues I documented earlier did not seem to be related to this update but once I did get them resolved, I found an incompatibility in the package versions for Apple M1. I could get it to work by updating `sass-embedded` to the latest beta version as directed in this issue comment: https://github.com/sass/embedded-host-node/issues/66#issuecomment-930581328

```
gem 'sass-embedded', '~> 1.0.0-beta.4'
```

Can you test on your machine if making that update affects your build adversely? Otherwise, I think this is working fine on my machine now as long as that gem is using the beta.4 version.

**Comment by bkjohnson** [comment]
After the changes from that commit and running `bundle install`, the server ran just fine.

**Review by k80bowman — APPROVED** [review]
I was able to get this running without even upgrading my local Ruby version, so that's good. We'll just need to make sure that everyone runs `bundle install` after this is merged and they pull this change down locally.

**Pull request merged**